Update introduction.stories.mdx (#897)
jpmorganchase/uitk: packages/theme/stories/introduction.stories.mdx

<Meta title="Documentation/Styles and Theming/Introduction" />

# Salt Theme

## Our approach

> The Salt design system is built from Vanilla CSS, which enables us to provide a custom and consistent set of styles. There are several benefits to using the Vanilla CSS framework:

Reviewer: ```suggestion
The Salt design system is built with Vanilla CSS, which enables us to provide a custom and consistent set of styles. There are several benefits to using Vanilla CSS:
```
```suggestion
The Salt design system is built using Vanilla CSS, which enables us to provide a custom and consistent set of styles. There are several benefits to using Vanilla CSS:
```

Either word is probably better, also vanilla CSS is not a framework

> - Salt has no third-party style dependencies, which means we are free to take our own approach to styling

Reviewer 2: ```suggestion
- Salt has no third-party style dependencies, which means we are free to take our own approach to styling.
```

- The Salt theme is built entirely using CSS custom properties.
- Vanilla CSS provides performance improvements by removing the runtime overhead common in CSS-in-JS solutions.

> - Vanilla CSS allows the Salt theme to be published and consumed independent of the component library–It can be used in contexts where Salt components are not able to be used, e.g. static code deployment within a CMS.

Reviewer 3: Our style guide recommends against using 'e.g.'. Suggest either the obvious 'for example' or '...such as environments with static code deployed within a CMS.'

Author: I've made the change, thanks.

}
```

## CSS levels

Salt styles are defined at two different levels, which operate together at runtime:

- **Theme styles** - consisting of CSS Variables \*
- **Component styles** - each component contributes its own stylesheet. Many of the CSS declarations reference CSS variables defined in the theme i.e. the variables will be defined at the theme level and used at the component level.

> \* Also includes one _utility_ class

Reviewer: Seems weird to mention this, I would probably omit it.

## Theme styles

The Salt theme is a collection of design tokens split into a three-tier hierarchy. The token tiers are:

1. **Foundation** - Foundation tokens are the primitive values for attributes like color, size, shadow and opacity.

> 2. **Palette** - semantically grouped foundational colors–neutral, navigate, interact –which allow a consistent look-and-feel when used by characteristics, rather than referencing the foundational tokens directly. For example, actionable, selectable, and editable characteristic tokens all use the 'interact' palette. If the theme needed to change the background color of interactive elements it would only have to be updated in one place, the palette, instead of within each characteristic.

Reviewer 3: Numbering should be used for sequential steps, these should be bullets. Also there seems to be a bit of a mix-up at the start of the Palette description, maybe a copy and paste error?

Author: Thanks Sion, I've updated the copy.
